### DISCUSSION TITLE
Idea: Extensible `neil add`

### DISCUSSION THREAD

**Discussion by rads** [babashka/neil#125] [Ideas]
Right now the only way to extend the `neil add` subcommand is by making a pull request. In contrast, `neil new` can be extended with any `deps.edn` lib.

Would it make sense for `neil add` to work in a similar way where the user can provide their own “template” if needed?

My main motivation for this is allowing the community to extend `neil` without having to mess around with its internal code. If there is a particularly popular or useful `neil add` extension, that could be a candidate for being included as a built-in.

```
# Built-in command (unqualified name)
$ neil add shadow-cljs

# External command (qualified name)
$ neil add io.github.rads/shadow-cljs
```

**Comment by borkdude** [comment]
I like the idea!
